src/StakingNode.sol

```diff
@@ -399,56 +463,90 @@ contract StakingNode is IStakingNode, StakingNodeEvents, ReentrancyGuardUpgradea
+     * @dev Call updateTotalETHStaked after this function
      * @param withdrawals Array of withdrawals to complete
-     * @param middlewareTimesIndexes Array of middleware times indexes
      */
     function completeQueuedWithdrawalsAsShares(
-        IDelegationManager.Withdrawal[] calldata withdrawals,
-        uint256[] calldata middlewareTimesIndexes
+        IDelegationManager.Withdrawal[] calldata withdrawals
     ) external onlyDelegator onlyWhenSynchronized {
-        uint256 totalWithdrawalAmount = 0;
+
+        syncQueuedShares();
+
+        uint256 totalWithdrawableShares = 0;
+
+        IDelegationManager delegationManager = stakingNodesManager.delegationManager();
 
         // Create empty tokens array since we're not receiving as tokens
-        IERC20[][] memory tokens = new IERC20[][](withdrawals.length);
+        IERC20V4[][] memory tokens = new IERC20V4[][](withdrawals.length);
         bool[] memory receiveAsTokens = new bool[](withdrawals.length);
 
         // Calculate total shares being withdrawn
         for (uint256 i = 0; i < withdrawals.length; i++) {
-            if (withdrawals[i].shares.length != 1 || withdrawals[i].strategies.length != 1 || withdrawals[i].strategies[0] != beaconChainETHStrategy) {
+            if (withdrawals[i].scaledShares.length != 1 || withdrawals[i].strategies.length != 1 || withdrawals[i].strategies[0] != beaconChainETHStrategy) {
                 revert InvalidWithdrawal();
             }
-            tokens[i] = new IERC20[](1);
+            tokens[i] = new IERC20V4[](1);
             receiveAsTokens[i] = false;
-            totalWithdrawalAmount += withdrawals[i].shares[0];
+            
+            totalWithdrawableShares += _decreaseQueuedSharesOnCompleteQueuedWithdrawalAndGetTotalWithdrawableShare(delegationManager, withdrawals[i]);
         }
 
-        IDelegationManager delegationManager = IDelegationManager(address(stakingNodesManager.delegationManager()));
-
         // Complete withdrawals with receiveAsTokens = false
-        delegationManager.completeQueuedWithdrawals(withdrawals, tokens, middlewareTimesIndexes, receiveAsTokens);
+        delegationManager.completeQueuedWithdrawals(
+            withdrawals, 
+            tokens, 
+            receiveAsTokens
+        );
 
-        // Decrease queued shares amount
-        queuedSharesAmount -= totalWithdrawalAmount;
+        emit CompletedQueuedWithdrawals(withdrawals, totalWithdrawableShares, 0);
+    }
 
-        emit CompletedQueuedWithdrawals(withdrawals, totalWithdrawalAmount, 0);
+    /**
+     * @notice Decreases the queued shares on complete queued withdrawals and returns the total withdrawable shares
+     * @param delegationManager The delegation manager
+     * @param withdrawal The withdrawal struct
+     * @return totalWithdrawableShareForWithdrawalRoot The total withdrawable shares for the withdrawal root
+     */
+    function _decreaseQueuedSharesOnCompleteQueuedWithdrawalAndGetTotalWithdrawableShare(
```

Comment:
`AndGetTotalWithdrawableShare` let's remove this from the name

and make a named return value like `returns (uint256 totalWithdrawableShare)` if you'd like to have that name clarity there, accompanied by the right natspec.


The return value is not normally mentioned in the function name, but normally named in the `returns (..)` syntax